sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java

```diff
@@ -735,11 +741,9 @@ private Object getCurrentKey() {
   }
 
   private void startBundle() {
+    timerBundleTracker.resetModifications();
```

Comment:
Register this with `PTransformRunnerFactory.Context#addResetFunction(...)` instead of resetting in `start Bundle`.